### PR TITLE
Ignore external errors

### PR DESCRIPTION
**Pull request by filipgolonka**
Rollbar saves some script errors, which are not from our domain. Solution recommended by Rollbar is to implement callback, which will check script domain in error stack trace.

See: https://gist.github.com/coryvirok/9478263

### CODE REVIEW

**Comment by mplachta** [comment]
:+1:

**Comment by ftrrtf** [comment]
@filipgolonka Thank you for MR, I will try check it out today or tomorrow.

**Comment by filipgolonka** [comment]
@ftrrtf thanks!

**Comment by wsromek** [comment]
:+1:

**Comment by filipgolonka** [comment]
@ftrrtf any progress?

**Comment by ftrrtf** [comment]
@filipgolonka But what if for js will be a separate some static domain, not origin? Maybe better to make the domain configurable?

**Comment by mplachta** [comment]
@ftrrtf :+1: :)

**Comment by filipgolonka** [comment]
:+1:, I will change it

**Comment by mplachta** [comment]
what about using assetic host configuration?

**Comment by ftrrtf** [comment]
@mplachta I think there are a lot of projects that do not use assetic

**Comment by mplachta** [comment]
OK, so what about introducing an extra setting:

`allowed_js_hosts`

By default (`allowed_js_hosts: ~`) it'll copy hosts from assetic bundle. But it'll also accept array of strings (hosts) like:

```yaml
allowed_js_hosts:
- this.is.my.first.host.com
- this.is.my.second.host.com
```

**Comment by filipgolonka** [comment]
@mplachta :+1:, that's exactly what I'm thinking about

**Comment by ftrrtf** [comment]
this option will be in `ftrrtf_rollbar.notifier.client` namespace, right?

**Comment by filipgolonka** [comment]
@ftrrtf of course.

**Comment by mplachta** [comment]
yeap :)

**Comment by ftrrtf** [comment]
Nice! :+1: 
Thank you for helping to improve the bundle ;)

**Comment by filipgolonka** [comment]
I decide not to merge it. I will add my fork to packagist and test changes from PR on production.

**Comment by mplachta** [comment]
And then PR it again, anyway @ftrrtf you can close this PR.

**Comment by filipgolonka** [comment]
@ftrrtf I decided to use my fork on production environment - after some hours it seems, that bundle stops reporting external errors using Rollbar.js. All changes are in this PR

**Comment by filipgolonka** [comment]
@ftrrtf any progress on it?

File: Twig/RollbarExtension.php [hunk 1]
```diff
@@ -91,15 +91,70 @@ public function getInitRollbarCode(array $context)
...
 
     /**
+     * @return string
+     */
+    protected function getCheckIgnoreConfig()
+    {
+        $allowedHosts = [];
+        if (
+            isset($this->notifierOptions['allowed_js_hosts']) &&
+            is_array($this->notifierOptions['allowed_js_hosts'])
```
Comment (ftrrtf) [inline Twig/RollbarExtension.php]: I guess that these checks and normalization is better to move into `DependencyInjection/Configuration`

Comment (mplachta) [inline Twig/RollbarExtension.php]: @ftrrtf :+1:

**Comment by filipgolonka** [comment]
@ftrrtf done, commits are squashed, please review.

**Comment by ftrrtf** [comment]
Looks good :+1: 

@mplachta let's review and merge

**Comment by mplachta** [comment]
:+1:

**Pull request merged**